src/core/DataBase_List.cpp

```diff
@@ -14,13 +14,45 @@
 
 using namespace std;
 
-List::List(){
+DataBase<string, deque<string>>::DataBase<string, deque<string>>(){
 }
 
-string List::lpop(string key) {
-    //I am not check for the key so it will throw a map exception if key does not exist
-    //thoughts?
-    deque<string> d = m.at(key);
+bool DataBase<string, deque<string>>::exist(const string& key) {
+    return (m.count(key) == 1);
+}
+
+   
+deque<string> DataBase<string, deque<string>>::get(const string& key){
+    if(!exist(key)){
+        throw "get: key does not exist in the list.";
+    }
+    return m[key];
+}
+
+bool DataBase<string, deque<string>>::remove(const string& key){
+    if(!exist(key)){
+        return false;
+    }
+    m.erase(key);
+    return true;
+}
+
+bool DataBase<string, deque<string>>::rename(const string& key, const string& newkey){
+    if(exist(newkey) || !exist(newkey)){
```

Comment:
I don't understand the purpose of this condition, it checks if the newkey exists or if it does not exist. If I understand correctly, this will always return false.